packages/schematics/update/update/index.ts

```diff
@@ -33,6 +33,9 @@ type PeerVersionTransform = string | ((range: string) => string);
 // We export it to allow for testing.
 export function angularMajorCompatGuarantee(range: string) {
   range = semver.validRange(range);
+  if (range === null) {
```

Comment:
I think in here, we should return the non mutated `range` as like this you are returning `null` which is also bypassing TypeScript type inference as this method shouldn't return `null`.

The problem here is the `range` is being mutated. I think the actual code should be something along;

```ts
export function angularMajorCompatGuarantee(range: string): string {
  let newRange = semver.validRange(range);
  if (!newRange) {
    return range;
  }

  let major = 1;
  while (!semver.gtr(major + '.0.0', newRange)) {
    major++;
    if (major >= 99) {
      // Use original range if it supports a major this high
      // Range is most likely unbounded (e.g., >=5.0.0)
      return newRange;
    }
  }

  // Add the major version as compatible with the angular compatible, with all minors. This is
  // already one major above the greatest supported, because we increment `major` before checking.
  // We add minors like this because a minor beta is still compatible with a minor non-beta.
  for (let minor = 0; minor < 20; minor++) {
    newRange += ` || ^${major}.${minor}.0-alpha.0 `;
  }

  return semver.validRange(newRange) || range;
}
```

Reply:
agreed, changed it 
Sorry for missing your reply.